functions/go/set-standard-labels/transformer/set-standard-labels.go

```diff
@@ -0,0 +1,85 @@
+package transformer
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-labels/setlabels"
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var _ fn.Runner = &SetStandardLabels{}
+
+type SetStandardLabels struct {
+	forDeployment string
+	// TBD: validate whether the recommended labels from the ResourceList.Items are set as expected.
+	validateOnly bool
+}
+
+// FnConfigFromLabel creates a FunctionConfig object for `set-label` KRM function based on the recommended labels `data`.
+func FnConfigFromLabels(data map[string]string) *fn.KubeObject {
+	fnConfig := fn.NewEmptyKubeObject()
+	fnConfig.SetKind(SetLabelFnKind)
+	fnConfig.SetAPIVersion("v1")
+	fnConfig.SetName(SetLabelFnName)
+	fnConfig.SetAnnotation(fn.KptLocalConfig, "true")
+	fnConfig.SetNestedStringMap(data, "data")
+	return fnConfig
+}
+
+// RecommendedLabels find out the recommended labels for the given package.
+func (r *SetStandardLabels) RecommendedLabels(items fn.KubeObjects) (map[string]string, error) {
+	data := map[string]string{}
+
+	// Get root package name from Kptfile
+	kptfile := items.GetRootKptfile()
+	if kptfile == nil {
+		return data, fmt.Errorf("no `Kptfile` are found. This function is for kpt pacakge. please run `kpt pkg init`")
+	}
+	pkgName := kptfile.GetName()
+
+	// Assign blueprint pkg name to "app.kubernetes.io/name" and deployment pkg name to "app.kubernetes.io/instance"
```

Comment:
When I was exploring argocd, this label was being used by `argocd` also for apps. Would be good to check if how well will it will play with argocd etc given that most of the users we are seeing are using kpt with argocd.